Microsoft.NET.Build.Containers/ContainerHelpers.cs

```diff
@@ -136,11 +136,28 @@ public static bool NormalizeImageName(string containerImageName, [NotNullWhen(fa
             if (Char.IsUpper(containerImageName, 0))
             {
                 containerImageName = Char.ToLowerInvariant(containerImageName[0]) + containerImageName[1..];
-            } else if (!Char.IsLetterOrDigit(containerImageName, 0)) {
+            }
+            else if (!Char.IsLetterOrDigit(containerImageName, 0))
+            {
                 throw new ArgumentException("The first character of the image name must be a lowercase letter or a digit.");
             }
             normalizedImageName = imageNameCharacters.Replace(containerImageName, "-");
             return false;
         }
     }
+
+    public static bool TryParsePort(string input, [NotNullWhen(true)] out Port? port)
+    {
+        var parts = input.Split('/');
+        Port? p =
+            parts switch
+            {
+                [var portNumber, var type] when int.TryParse(portNumber, out var portInt)
```

Comment:
I appreciate you using the newer c# features!

Thinking out loud as I make sense of it:

You switch over `parts` which is a list of items split on `/`.

If `parts` is two elements long, parse them and return a new Port if you could parse.
If you got a single element, parse the port and default to tcp.
Whatever else you got, abandon all hope.

The first case is hard to read if it's you first time looking it over, but no big deal once you learn it. I like it.

Reply:
This actually looks really nice in F# because patterns compose in a way that they currently don't in C#:

```fsharp
match p.Split("/") with
| [ Int portNum; Enum<PortType> portTye ] -> ...
| [Int portNum ] -> ...
| _ -> ...
```
you can nest patterns inside other patterns easily, including user-defined patterns (since a pattern is just a function).